contentgrid-spring-data-rest/src/main/java/org/springframework/data/rest/webmvc/DelegatingRepositoryPropertyReferenceController.java

```diff
@@ -0,0 +1,279 @@
+            }
+
+            // single-valued association
+            return prop.mapValue(target -> {
+                        var idAccessor = prop.entity.getIdentifierAccessor(target);
+                        if (propertyId.equals(Objects.requireNonNull(idAccessor.getIdentifier()).toString())) {
+                            // the property-id matches
+                            return target;
+                        }
+
+                        // if the propertyId does not match the linked resource, filter out by returning null
+                        return null;
+                    })
+                    .map(target -> this.selfLinkProvider.createSelfLinkFor(prop.getPropertyType(), target))
+                    .map(link -> link.getTemplate().expand())
+                    .map(uri -> ResponseEntity.status(HttpStatus.FOUND).location(uri).build())
+                    .orElse(ResponseEntity.notFound().build());
+
+        };
+
+        return doWithReferencedProperty(repoRequest, id, property, handler, HttpMethod.GET);
+    }
+
+    @RequestMapping(value = BASE_MAPPING, method = GET, produces = TEXT_URI_LIST_VALUE)
+    public ResponseEntity<RepresentationModel<?>> followPropertyReferenceCompact(RootResourceInformation repoRequest,
+            @BackendId Serializable id, @PathVariable String property, @RequestHeader HttpHeaders requestHeaders,
+            PersistentEntityResourceAssembler assembler) throws Exception {
+
+        return this.delegate.followPropertyReferenceCompact(repoRequest, id, property, requestHeaders, assembler);
+    }
+
+    @RequestMapping(value = BASE_MAPPING, method = {PATCH, PUT, POST},
+            consumes = {MediaType.APPLICATION_JSON_VALUE, SPRING_DATA_COMPACT_JSON_VALUE, TEXT_URI_LIST_VALUE})
+    public ResponseEntity<? extends RepresentationModel<?>> createPropertyReference(
+            RootResourceInformation resourceInformation, HttpMethod requestMethod,
+            @RequestBody(required = false) CollectionModel<Object> incoming, @BackendId Serializable id,
+            @PathVariable String property) throws Exception {
+
+        return this.delegate.createPropertyReference(resourceInformation, requestMethod, incoming, id, property);
+    }
+
+    @RequestMapping(value = BASE_MAPPING + "/{propertyId}", method = DELETE)
+    public ResponseEntity<RepresentationModel<?>> deletePropertyReferenceId(RootResourceInformation repoRequest,
+            @BackendId Serializable backendId, @PathVariable String property, @PathVariable String propertyId)
+            throws Exception {
+
+        return this.delegate.deletePropertyReferenceId(repoRequest, backendId, property, propertyId);
+    }
+
+    @ExceptionHandler
+    public ResponseEntity<Void> handle(
+            RepositoryPropertyReferenceController.HttpRequestMethodNotSupportedException exception) {
+        return exception.toResponse();
+    }
+
+    private ResponseEntity<?> doWithReferencedProperty(
```

Comment:
Maybe add a comment that this is copied from spring-data-rest.

If we hit some issues later-on, we know to compare it to spring-data-rest for changes, because perhaps some implementation details have changed internally.

Reply:
Done